src/outlier_hub/datasets/ham10k/preprocessor.py

```diff
@@ -0,0 +1,111 @@
+import csv
+import os
+import tempfile
+from typing import Tuple, List
+
+import h5py
+import matplotlib.image as mpimg
+import numpy as np
+from data_stack.io.resources import StreamedResource, ResourceFactory
+from data_stack.io.storage_connectors import StorageConnector
+
+
+class HAMPreprocessor:
+
+    def __init__(self, storage_connector: StorageConnector):
+        self.storage_connector = storage_connector
+        # TODO right now only one split is needed
+        self.split_names = ["raw"]
+
+    # def preprocess(self, img_resource: StreamedResource, metadata_resource: StreamedResource):
+    def preprocess(self, identifier_set: Tuple[str, str, str]) -> StreamedResource:
+        with tempfile.TemporaryFile() as temp_file:
+            with h5py.File(temp_file, 'w') as h5py_file:
+                for split_name in self.split_names:
+                    self._preprocess_split(h5py_file,
+                                           split_name,
+                                           temp_file,
+                                           identifier_set)
+                h5py_file.flush()
+                temp_file.flush()
+            streamed_resource = ResourceFactory.get_resource(identifier_set[0], temp_file)
+            self.storage_connector.set_resource(identifier_set[0], streamed_resource)
+            streamed_resource = self.storage_connector.get_resource(identifier_set[0])
+        return streamed_resource
+
+    def _preprocess_split(self,
+                          h5py_file: h5py.File,
+                          split_name: str,
+                          temporary_file,
+                          identifier_set: Tuple[str, str, str]):
+        split_samples, split_targets = self._get_raw_dataset_split(split_name, identifier_set)
+
+        sample_location = os.path.join(split_name, "samples")
+        target_location = os.path.join(split_name, "targets")
+
+        sample_dset = h5py_file.create_dataset(sample_location, shape=(len(split_samples), 450, 600, 3), dtype=int)
+
+        # h5py cannot save np.ndarrays with strings by default, costum dtype must be created
+        utf8_type = h5py.string_dtype('utf-8')
+
+        target_dset = h5py_file.create_dataset(target_location, shape=(len(split_targets), 8,),dtype=utf8_type)
+
+        for cnt, sample in enumerate(split_samples):
+            sample_dset[cnt:cnt + 1, :, :] = mpimg.imread(sample)
+
+        for cnt, target in enumerate(split_targets):
+            target_dset[cnt] = target
+
+    def _get_raw_dataset_split(self,split_name: str,identifier_set: Tuple[str, ...]) -> Tuple[List[str], List[
```

Comment:
are these ... pep8 conform? Never seen them before.

Reply:
I updated the signature due the proper arguments implementation. Does that contain the point what you mention here?
Otherwise I am not sure what you mean and pycharm does not mention PEP8 violations in particular.